### ISSUE TITLE
Weird MathJax output when using TeX-AMS_CHTML-full

### ISSUE THREAD

**Issue by dotlambda** [hakimel/reveal.js#1924]
I have no idea why the math displayed in reveal.js using MathJax's CommonHTML output looks like that:
![2017-06-18-020850_1920x1080_scrot](https://user-images.githubusercontent.com/6806011/27260917-85a4e22a-5427-11e7-8a2a-1e2b4cffef51.png)
The source code for this is:
```html
<!DOCTYPE html>
<html>
<head>
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <link rel="stylesheet" href="reveal.js/css/theme/black.css" id="theme">
</head>
<body>
  <div class="reveal">
    <div class="slides">
<section id="conclusion" class="slide level1">
<h1>Conclusion</h1>
<ul>
<li>And the answer is…</li>
<li><span class="math inline">\(f(x)=\sum_{n=0}^\infty\frac{f^{(n)}(a)}{n!}(x-a)^n\)</span></li>
</ul>
</section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      Reveal.initialize({
        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_CHTML-full'
        },

        dependencies: [
          { src: 'reveal.js/plugin/math/math.js', async: true },
        ]
      });
    </script>
    </body>
</html>
```

**Comment by benni336** [comment]
Is this an reveal.js issue?

**Comment by dotlambda** [comment]
Yes, because when using MathJax CHTML output on a blank website without reveal.js, it works just fine.

**Comment by NuclearCookie** [comment]
The issue comes from the way that reveal.js does the reset styles.
the reset only applies to subclasses of the div with a class `reveal`. 
Sadly, this adds specificity for the css parser. It's more specific to set a property on `.reveal span` than on `.MJX-char` for example. So it overrides the font usage. 

A good solution for that would be to reset _all_ html elements instead of only the ones in the reveal div.

**Comment by Beramos** [comment]
I've come across the same issue and ended up with the same solution as @mbotsch. However, the solution proposed by @NuclearCookie would be a more permanent solution for the future usage of reveal.js + MathJax

**Comment by mbotsch** [comment]
I use Reveal for teaching math-heavy courses and found the CommonHTML output of MathJax to be more consistent between browsers and between operating systems than the old HTML output. Since the default output of MathJax is CommonHTML since version 2.7, it would be really cool if Reveal could switch to that.

Here's the solution that worked for me:

1) Remove the style reset for `.reveal span` from reveal.css. I agree that the solution of @NuclearCookie is probably better.

2) Even then I had problems that with fonts being way too small when I set Reveal's viewDistance to a small value (which I have to for some performance reasons) and then switch to a slide with equations that becomes visible but was not visible at the time of (re)loading the slides. This problem is caused by MathJax trying to adjust its font size in order to match the text's actual, rendered font size. I just disabled the font size adjustment, which also resolved the problem that the same equation on two successive slides might slightly change its size. I had to adjust the MathJax config like this:

MathJax.Hub.Config({
    jax: ["input/TeX","output/CommonHTML"],
    "CommonHTML": { matchFontHeight: false }
});

With these two adjustments, which could probably be integrated easily, the CommonHTML works like a charme.

**Comment by eddieschoute** [comment]
[KaTeX](https://katex.org/) fonts are also being overriden by `.reveal span`. This results in ugly typesetting.

**Comment by csimaoferreira** [comment]
This solution appears to work, where the font size is consistent at first load.

math: {
	mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.6/MathJax.js',
	config: 'TeX-MML-AM_CHTML-full', 
	// pass other options into `MathJax.Hub.Config()`
		TeX: { Macros: { RR: "{\\bf R}" } ,  equationNumbers: { autoNumber: "AMS" }},
		jax: ["input/TeX","output/CommonHTML"],"CommonHTML": { matchFontHeight: false, scale: 120 }
},
